### PR TITLE
trivial: ci: add a test that the snap works

### PR DESCRIPTION
**Pull request by superm1**
Use the *Integrated Webcam* test device to check that the snap can update devices using the local vendor directory.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: .github/workflows/snap.yml [hunk 1]
```diff
@@ -34,6 +34,37 @@ jobs:
...
+      with:
+        name: snap
+    - id: install-snap
+      run: |
+        sudo snap install --dangerous ${{ needs.build-snap.outputs.snap_name }}
+        sudo mkdir -p /var/snap/fwupd/common/var/etc/fwupd
+        sed "s,^DisabledPlugins=.*,DisabledPlugins=," /snap/fwupd/current/etc/fwupd/fwupd.conf | sudo tee /var/snap/fwupd/common/var/etc/fwupd/fwupd.conf
+    - id: build-cab
+      run: |
+        gcab --create  --nopath fakedevice124.cab data/installed-tests/fakedevice124.*
```
Comment (hughsie) [inline .github/workflows/snap.yml:60]: `fwupdmgr firmware-build` perhaps?

File: .github/workflows/snap.yml [hunk 2]
```diff
@@ -34,6 +34,37 @@ jobs:
...
+        sudo apt install -y snapd gcab
+    - uses: actions/download-artifact@v4
+      id: download
+      with:
+        name: snap
+    - id: install-snap
+      run: |
+        sudo snap install --dangerous ${{ needs.build-snap.outputs.snap_name }}
+        sudo mkdir -p /var/snap/fwupd/common/var/etc/fwupd
+        sed "s,^DisabledPlugins=.*,DisabledPlugins=," /snap/fwupd/current/etc/fwupd/fwupd.conf | sudo tee /var/snap/fwupd/common/var/etc/fwupd/fwupd.conf
```
Comment (hughsie) [inline .github/workflows/snap.yml:57]: i wondered if we should have a fwupdtool/fwupdmgr command to do this; it's something we seem to do a lot

Comment (superm1) [reply .github/workflows/snap.yml:57]: We can probably do it with `fwupdmgr modify-config DisabledPlugins`.

File: .github/workflows/snap.yml (hunk 1 above)
Comment (superm1) [reply .github/workflows/snap.yml:60]: Did you mean `fwupdtool`, or actually port `firmware-build` to `fwupdmgr`?

Comment (hughsie) [reply .github/workflows/snap.yml:60]: Bleh, fwupdtool indeed, sorry.

File: .github/workflows/snap.yml (hunk 2 above)
Comment (superm1) [reply .github/workflows/snap.yml]: Hmm; modify-config is broken...  Look what a few runs are doing:

```
[fwupd]
DisabledPlugins=test;test_ble
OnlyTrusted=true
AllowEmulation=false
DisabledPlugins=
OnlyTrusted=true
AllowEmulation=false
DisabledPlugins=test;test_ble
OnlyTrusted=true
AllowEmulation=false
DisabledPlugins=test_ble
OnlyTrusted=true
AllowEmulation=false
```

Comment (superm1) [reply .github/workflows/snap.yml]: Given that bug I think should leave as it for introducing the test and clean up later.

File: .github/workflows/snap.yml (hunk 1 above)
Comment (hughsie) [reply .github/workflows/snap.yml:61]: I got it double wrong -- what I meant was `fwupdtool build-cabinet ARCHIVE FIRMWARE METAINFO`

File: .github/workflows/snap.yml (hunk 2 above)
Comment (hughsie) [reply .github/workflows/snap.yml]: Eeek, can you file it and I'll fix it on Monday please.

**Review by hughsie — APPROVED** [review]

**Pull request merged**